### PR TITLE
fix(async-csv): Prevent direct object reference on data exports

### PR DESCRIPTION
**Pull request by leeandher**
See VULN-4

This PR will prevent directly being able to download data exports with their ID regardless of the organization they belong to. 

**Before**
_Invalid org for this export_
Screen Shot 2021-11-10 at 11 36 21 AM

**After**
_Invalid org for this export_
Screen Shot 2021-11-10 at 12 26 37 PM
image

_Valid org for this export_
Screen Shot 2021-11-10 at 12 27 11 PM

**TODO**
 - [x] Add tests for this behavior

### CODE REVIEW

**Review by AniketDas-Tekky — COMMENTED** [review]
Can you add a test?

File: src/sentry/data_export/endpoints/data_export_details.py [hunk 1]
```diff
@@ -25,7 +25,9 @@ def get(self, request, organization, **kwargs):
             return Response(status=404)
 
         try:
-            data_export = ExportedData.objects.get(id=kwargs["data_export_id"])
+            data_export = ExportedData.objects.get(
```
Comment (mgaeta) [inline src/sentry/data_export/endpoints/data_export_details.py:28]: We should have the minimum number of lines in the try/catch:
```py
try:
    data_export = ExportedData.objects.get(
        id=kwargs["data_export_id"], organization=organization
    )
except ExportedData.DoesNotExist:
    return Response(status=404)
...
```

File: src/sentry/data_export/endpoints/data_export_details.py [hunk 2]
```diff
@@ -25,21 +27,21 @@ def get(self, request, organization, **kwargs):
...
+            data_export = ExportedData.objects.get(id=data_export_id, organization=organization)
         except ExportedData.DoesNotExist:
             return Response(status=404)
+        # Check data export permissions
+        if data_export.query_info.get("project"):
+            project_ids = map(int, data_export.query_info.get("project", []))
+            projects = Project.objects.filter(organization=organization, id__in=project_ids)
+            if any(p for p in projects if not request.access.has_project_access(p)):
+                raise PermissionDenied(
+                    detail="You don't have access to some of the data this export contains."
```
Comment (dashed) [inline src/sentry/data_export/endpoints/data_export_details.py:39]:
```suggestion
                    detail="You don't have permissions to some of the data this export contains."
```

Comment (leeandher) [reply src/sentry/data_export/endpoints/data_export_details.py]: thanks! i changed it to permissions and fixed the grammar

**Review by dcramer — APPROVED** [review]
lgtm from security side

**Review by mgaeta — APPROVED** [review]

**Pull request merged**